src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -257,9 +257,172 @@ public Task<GitDiff> GitDiffAsync(string repoUri, string baseVersion, string tar
+                                    _logger.LogDebug($"Fetching {string.Join(";", refSpecs)} from {r.Url} in {repoDir}");
+                                    try
+                                    {
+                                        LibGit2Sharp.Commands.Fetch(localRepo, r.Name, refSpecs, new LibGit2Sharp.FetchOptions(), $"Fetching from {r.Url}");
+                                    }
+                                    catch
+                                    {
+                                        _logger.LogWarning($"Fetching failed, are you offline or missing a remote?");
+                                    }
+                                }
+                                _logger.LogDebug($"After fetch, attempting to checkout {commit} in {repoDir}");
+                                try
+                                {
+                                    LibGit2Sharp.Commands.Checkout(localRepo, commit, checkoutOptions);
+                                }
+                                catch
+                                {
+                                    _logger.LogDebug($"Couldn't checkout {commit} as a commit after fetch.  Attempting to resolve as a treeish.");
+                                    string resolvedReference = ParseReference(localRepo, commit, _logger);
+                                    if (resolvedReference != null)
+                                    {
+                                        _logger.LogDebug($"Resolved {commit} to {resolvedReference}, attempting to check out");
+                                        LibGit2Sharp.Commands.Checkout(localRepo, resolvedReference, checkoutOptions);
+                                    }
+                                }
+
+                                if (force)
+                                {
+                                    CleanRepoAndSubmodules(localRepo, _logger);
+                                }
+                            }
+                            catch   // Most likely network exception, could also be no remotes.  We can't do anything about any error here.
+                            {
+                                _logger.LogError($"After fetch, still couldn't find commit or treeish {commit} in {repoDir}.  Are you offline or missing a remote?");
+                                throw;
+                            }
+                        }
+                    }
+                }
+                catch (Exception exc)
+                {
+                    throw new Exception($"Something went wrong when checking out {commit} in {repoDir}", exc);
+                }
+            }
+        }
+
+        private static void CleanRepoAndSubmodules(LibGit2Sharp.Repository repo, ILogger log)
+        {
+            log.LogDebug($"Beginning clean of {repo.Info.WorkingDirectory} and {repo.Submodules.Count()} submodules");
```

Comment:
>log.LogDebug [](start = 12, length = 12)

log.BeginScope

Reply:
Changed as suggested.

---
In reply to: [275517872](https://github.com/dotnet/arcade-services/pull/326#discussion_r275517872) [](ancestors = 275517872)